### PR TITLE
Use spring5

### PR DESCRIPTION
**Pull request by taitori**

### CODE REVIEW

**Comment by taitori** [comment]
@tkruse Could you help review my change?

### Context
We want to update some library versions used in proctor to avoid vulnerable versions. This PR tries to update `spring` and `jackson`.
Reading through the current build configuration and having some conversation with @isozuka , I learned that 1: proctor wants to keep using el7 (`tomcat 7`) and 2: it shades el7 dependencies inside `proctor-common`.

I specifically would like you to check whether updating the `spring` version is enough or not. Because [spring 5 requires servlet 3.1+](https://docs.spring.io/spring-framework/docs/current/reference/html/overview.html#overview-history) and [tomcat 7 only supports servlet 3.0](https://tomcat.apache.org/whichversion.html), I'm concerned that forcing to use tomcat 7 causes some problems.

File: pom.xml [hunk 1]
```diff
@@ -73,10 +73,10 @@
         <!--
         the versions defined in javadev
         -->
-        <spring.version>4.3.18.RELEASE</spring.version>
+        <spring.version>5.2.22.RELEASE</spring.version>
```
Comment (tkruse) [inline pom.xml:76]: as mentioned in chat, I suspect just this change might break proctor-webapp and proctor-demo. The actual problem to be fixed can be fixed in indeeds internal libraries with a gradle constraint on spring5

File: proctor-consumer/pom.xml [hunk 2]
```diff
@@ -32,13 +32,6 @@
             <artifactId>proctor-common</artifactId>
             <version>${project.version}</version>
         </dependency>
-        <dependency>
-            <!-- for jasper-el -->
-            <groupId>${project.groupId}</groupId>
-            <artifactId>proctor-tomcat-deps</artifactId>
-            <version>${project.version}</version>
-            <type>pom</type>
-        </dependency>
```
Comment (tkruse) [inline proctor-consumer/pom.xml:41]: I cannot see or remember if this dependency is required. If we don't see usage of this, then it's probably safe to remove, it seems it was added in an early version of proctor

File: pom.xml (hunk 1 above)
Comment (taitori) [reply pom.xml:76]: Thanks for pointing it out. I'll try to run proctor-webapp and proctor-demo. If I detect issues and find them very hard to fix, I'm going to use gradle constraint as you suggested.

File: proctor-consumer/pom.xml (hunk 2 above)
Comment (taitori) [reply proctor-consumer/pom.xml:41]: Got it.

File: pom.xml (hunk 1 above)
Comment (taitori) [reply pom.xml:76]: Hmm, I noticed even the master branch cannot run `proctor-webapp`. It failed with the following errors:
```
$ java -Dconfig.dir="$PWD/proctor-webapp/example-apache-config"  -jar proctor-webapp/target/dependency/webapp-runner.jar --context-xml proctor-webapp/example-apache-config/proctor-webapp.xml --expand-war proctor-webapp/target/proctor-webapp-9999-SNAPSHOT.warExpanding proctor-webapp-9999-SNAPSHOT.war into /home/taitori/indeed/proctor/target/tomcat.8080/webapps/expanded
Adding Context  for /home/taitori/indeed/proctor/target/tomcat.8080/webapps/expanded
Using context config: proctor-webapp/example-apache-config/proctor-webapp.xml
May 31, 2022 5:07:35 PM org.apache.coyote.AbstractProtocol init
INFO: Initializing ProtocolHandler ["http-nio-8080"]
May 31, 2022 5:07:35 PM org.apache.tomcat.util.net.NioSelectorPool getSharedSelector
INFO: Using a shared selector for servlet write/read
May 31, 2022 5:07:35 PM org.apache.catalina.core.StandardService startInternal
INFO: Starting service Tomcat
May 31, 2022 5:07:35 PM org.apache.catalina.core.StandardEngine startInternal
INFO: Starting Servlet Engine: Apache Tomcat/7.0.40
May 31, 2022 5:07:35 PM org.apache.catalina.startup.SetContextPropertiesRule begin
WARNING: [SetContextPropertiesRule]{Context} Setting property 'debug' to '5' did not find a matching property.
May 31, 2022 5:07:35 PM org.apache.catalina.startup.ContextConfig getDefaultWebXmlFragment
INFO: No global web.xml found
May 31, 2022 5:07:36 PM org.apache.catalina.startup.ContextConfig processAnnotationsJar
SEVERE: Unable to process Jar entry [META-INF/versions/9/module-info.class] from Jar [jar:file:/home/taitori/indeed/proctor/target/tomcat.8080/webapps/expanded/WEB-INF/lib/log4j-api-2.17.1.jar!/] for annotations
org.apache.tomcat.util.bcel.classfile.ClassFormatException: Invalid byte tag in constant pool: 19
        at org.apache.tomcat.util.bcel.classfile.Constant.readConstant(Constant.java:133)
        at org.apache.tomcat.util.bcel.classfile.ConstantPool.<init>(ConstantPool.java:60)
        at org.apache.tomcat.util.bcel.classfile.ClassParser.readConstantPool(ClassParser.java:209)
        at org.apache.tomcat.util.bcel.classfile.ClassParser.parse(ClassParser.java:119)
        at org.apache.catalina.startup.ContextConfig.processAnnotationsStream(ContextConfig.java:2125)
        at org.apache.catalina.startup.ContextConfig.processAnnotationsJar(ContextConfig.java:2001)
        at org.apache.catalina.startup.ContextConfig.processAnnotationsUrl(ContextConfig.java:1967)
        at org.apache.catalina.startup.ContextConfig.processAnnotations(ContextConfig.java:1952)
        at org.apache.catalina.startup.ContextConfig.webConfig(ContextConfig.java:1326)
        at org.apache.catalina.startup.ContextConfig.configureStart(ContextConfig.java:878)
        at org.apache.catalina.startup.ContextConfig.lifecycleEvent(ContextConfig.java:369)
        at org.apache.catalina.util.LifecycleSupport.fireLifecycleEvent(LifecycleSupport.java:119)
        at org.apache.catalina.util.LifecycleBase.fireLifecycleEvent(LifecycleBase.java:90)
        at org.apache.catalina.core.StandardContext.startInternal(StandardContext.java:5269)
        at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:150)
        at org.apache.catalina.core.ContainerBase$StartChild.call(ContainerBase.java:1559)
        at org.apache.catalina.core.ContainerBase$StartChild.call(ContainerBase.java:1549)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)

May 31, 2022 5:07:37 PM org.apache.catalina.core.ContainerBase startInternal
SEVERE: A child container failed during start
java.util.concurrent.ExecutionException: org.apache.catalina.LifecycleException: Failed to start component [StandardEngine[Tomcat].StandardHost[localhost].StandardContext[]]
        at java.util.concurrent.FutureTask.report(FutureTask.java:122)
        at java.util.concurrent.FutureTask.get(FutureTask.java:192)
        at org.apache.catalina.core.ContainerBase.startInternal(ContainerBase.java:1123)
        at org.apache.catalina.core.StandardHost.startInternal(StandardHost.java:800)
        at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:150)
        at org.apache.catalina.core.ContainerBase$StartChild.call(ContainerBase.java:1559)
        at org.apache.catalina.core.ContainerBase$StartChild.call(ContainerBase.java:1549)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)
Caused by: org.apache.catalina.LifecycleException: Failed to start component [StandardEngine[Tomcat].StandardHost[localhost].StandardContext[]]
        at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:154)
        ... 6 more
Caused by: java.lang.NoClassDefFoundError: freemarker/template/ObjectWrapper
        at java.lang.Class.getDeclaredFields0(Native Method)
        at java.lang.Class.privateGetDeclaredFields(Class.java:2583)
        at java.lang.Class.getDeclaredFields(Class.java:1916)
        at org.apache.catalina.util.Introspection.getDeclaredFields(Introspection.java:106)
        at org.apache.catalina.startup.WebAnnotationSet.loadFieldsAnnotation(WebAnnotationSet.java:261)
        at org.apache.catalina.startup.WebAnnotationSet.loadApplicationServletAnnotations(WebAnnotationSet.java:140)
        at org.apache.catalina.startup.WebAnnotationSet.loadApplicationAnnotations(WebAnnotationSet.java:67)
        at org.apache.catalina.startup.ContextConfig.applicationAnnotationsConfig(ContextConfig.java:405)
        at org.apache.catalina.startup.ContextConfig.configureStart(ContextConfig.java:881)
        at org.apache.catalina.startup.ContextConfig.lifecycleEvent(ContextConfig.java:369)
        at org.apache.catalina.util.LifecycleSupport.fireLifecycleEvent(LifecycleSupport.java:119)
        at org.apache.catalina.util.LifecycleBase.fireLifecycleEvent(LifecycleBase.java:90)
        at org.apache.catalina.core.StandardContext.startInternal(StandardContext.java:5269)
        at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:150)
        ... 6 more
Caused by: java.lang.ClassNotFoundException: freemarker.template.ObjectWrapper
        at org.apache.catalina.loader.WebappClassLoader.loadClass(WebappClassLoader.java:1714)
        at org.apache.catalina.loader.WebappClassLoader.loadClass(WebappClassLoader.java:1559)
        ... 20 more

May 31, 2022 5:07:37 PM org.apache.catalina.core.ContainerBase startInternal
SEVERE: A child container failed during start
java.util.concurrent.ExecutionException: org.apache.catalina.LifecycleException: Failed to start component [StandardEngine[Tomcat].StandardHost[localhost]]
        at java.util.concurrent.FutureTask.report(FutureTask.java:122)
        at java.util.concurrent.FutureTask.get(FutureTask.java:192)
        at org.apache.catalina.core.ContainerBase.startInternal(ContainerBase.java:1123)
        at org.apache.catalina.core.StandardEngine.startInternal(StandardEngine.java:302)
        at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:150)
        at org.apache.catalina.core.StandardService.startInternal(StandardService.java:443)
        at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:150)
        at org.apache.catalina.core.StandardServer.startInternal(StandardServer.java:732)
        at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:150)
        at org.apache.catalina.startup.Tomcat.start(Tomcat.java:335)
        at webapp.runner.launch.Main.main(Main.java:211)
Caused by: org.apache.catalina.LifecycleException: Failed to start component [StandardEngine[Tomcat].StandardHost[localhost]]
        at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:154)
        at org.apache.catalina.core.ContainerBase$StartChild.call(ContainerBase.java:1559)
        at org.apache.catalina.core.ContainerBase$StartChild.call(ContainerBase.java:1549)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)
Caused by: org.apache.catalina.LifecycleException: A child container failed during start
        at org.apache.catalina.core.ContainerBase.startInternal(ContainerBase.java:1131)
        at org.apache.catalina.core.StandardHost.startInternal(StandardHost.java:800)
        at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:150)
        ... 6 more

Exception in thread "main" org.apache.catalina.LifecycleException: Failed to start component [StandardServer[-1]]
        at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:154)
        at org.apache.catalina.startup.Tomcat.start(Tomcat.java:335)
        at webapp.runner.launch.Main.main(Main.java:211)
Caused by: org.apache.catalina.LifecycleException: Failed to start component [StandardService[Tomcat]]
        at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:154)
        at org.apache.catalina.core.StandardServer.startInternal(StandardServer.java:732)
        at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:150)
        ... 2 more
Caused by: org.apache.catalina.LifecycleException: Failed to start component [StandardEngine[Tomcat]]
        at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:154)
        at org.apache.catalina.core.StandardService.startInternal(StandardService.java:443)
        at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:150)
        ... 4 more
Caused by: org.apache.catalina.LifecycleException: A child container failed during start
        at org.apache.catalina.core.ContainerBase.startInternal(ContainerBase.java:1131)
        at org.apache.catalina.core.StandardEngine.startInternal(StandardEngine.java:302)
        at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:150)
        ... 6 more
```

**Comment by taitori** [comment]
Close this MR as we need more effort to make the things work correctly. The current blockers are:
* Runtime error in proctor-webapp
* Code change for the https://github.com/indeedeng/proctor-demo and other projects using this repository

**Pull request closed without merging**